benchmarks/instruction_counts/execution/runner.py

```diff
@@ -0,0 +1,240 @@
+import math
+import multiprocessing
+import subprocess
+import textwrap
+import threading
+import time
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from execution.work import PYTHON_CMD, SHELL, InProgress, WorkOrder
+from worker.main import WorkerFailure, WorkerOutput
+
+
+CPU_COUNT: int = multiprocessing.cpu_count()
+
+
+class WorkerFailed(Exception):
+    """Raised in the main process when a worker failure is detected."""
+    def __init__(self, wrapped_trace: Optional[str] = None) -> None:
+        self.wrapped_trace: Optional[str] = wrapped_trace
+        super().__init__()
+
+
+class CorePool:
+    def __init__(self, min_core_id: int, max_core_id: int) -> None:
+        assert min_core_id >= 0
+        assert max_core_id >= min_core_id
+        assert max_core_id < CPU_COUNT
+
+        self._min_core_id: int = min_core_id
+        self._max_core_id: int = max_core_id
+        self._num_cores = max_core_id - min_core_id + 1
+
+        self._available: List[bool] = [
+            True for _ in range(min_core_id, min_core_id + self._num_cores)]
+
+        self._reservations: Dict[str, Tuple[int, ...]] = {}
+        self._lock = threading.Lock()
+
+    def reserve(self, n: int) -> Optional[str]:
+        """Simple first-fit policy.
+
+        If successful, return a string for `taskset`. Otherwise, return None.
```

Comment:
Can you clarify that n is the number of CPUs we want to reserve in a contiguous block?

Reply:
Indeed. I added a docstring to `CorePool` explaining some of the reasoning.